### PR TITLE
handle rss and json for streams

### PR DESCRIPTION
**Pull request by wheresrhys**
@aintgoin2goa @i-like-robots @commuterjoy 

More than likely will need a major release to do this properly at some point, but as that will need some thought putting in to the vanities architecture it'll take some time. Important thing for now is to get this out to fix a production bug

### CODE REVIEW

**Comment by i-like-robots** [comment]
LGTM, thanks!

File: main.js [hunk 1]
```diff
@@ -13,14 +13,21 @@ let timeout;
...
 		let cacheItem = cache.retrieve(fromURL);
 		if(cacheItem){
 			return Promise.resolve(cacheItem);
 		}
 	}
+	const extension = (extensionRX.exec(fromURL) || [''])[0];
+
 
+	if (extension) {
+		fromURL = fromURL.replace(extensionRX, '');
```
Comment (aintgoin2goa) [inline main.js]: Surely you only need this line?  if it doesn't match it won't replace it

**Comment by aintgoin2goa** [comment]
:+1:

**Comment by matthew-andrews** [comment]
i think matt h found a different solution to this, i think we might not need to merge this?

**Comment by wheresrhys** [comment]
Yep. I think an optimal solution is still to be arrived at though.
fastft?format=rss should be supported by default I feel ( given the /fastft
vanity), but the API rather than the client should do the work

On Wednesday, 30 March 2016, Matt Andrews <notifications@github.com> wrote:

> i think matt h found a different solution to this, i think we might not
> need to merge this?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Financial-Times/n-url-management-api-read-client/pull/27#issuecomment-203603740>
>

-- 

------------------------------

*This email was sent by a company owned by Financial Times Group Limited 
("FT Group <http://aboutus.ft.com/corporate-information/#axzz3rajCSIAt>"), 
registered office at Number One Southwark Bridge, London SE1 9HL.  
Registered in England and Wales with company number 879531. This e-mail may 
contain confidential information. If you are not the intended recipient, 
please notify the sender immediately, delete all copies and do not 
distribute it further.  It could also contain personal views which are not 
necessarily those of the FT Group.  We may monitor outgoing or 
incoming emails as permitted by law.*

**Comment by matthew-andrews** [comment]
i'm not sure i understand — `?format=rss` is supported by default…?

**Comment by matthew-andrews** [comment]
ah do you mean the stripping off query strings should happen in `n-url-management-api-client` rather than in preflight/other consumers of it?

**Comment by wheresrhys** [comment]
can't remember what I meant exactly. Doesn't make much sense to me. I have a feeling I realised what you pointed out above just after posting

**Pull request closed without merging**